### PR TITLE
ADEN-8443 Disable skin slot when there is no roadblock

### PR DESCRIPTION
**Pull request by fraszczakszymon**

### CODE REVIEW

File: extensions/wikia/AdEngine3/src/setup.js [hunk 1]
```diff
@@ -194,7 +194,7 @@ function setupAdContext(wikiContext, isOptedIn = false, geoRequiresConsent = tru
 
 		if (slot instanceof Array) {
 			slotName = slot[0];
-		} else if (typeof slot === 'string') {
+		} else if (typeof slot === 'object' && slot.slotName) {
```
Comment (fraszczakszymon) [inline extensions/wikia/AdEngine3/src/setup.js:197]: 😱 :wtf:

Comment (jacekbj) [reply extensions/wikia/AdEngine3/src/setup.js]: Can you elaborate?

File: extensions/wikia/AdEngine3/src/templates/roadblock-config.js [hunk 2]
```diff
@@ -1,8 +1,13 @@
...
+
 export function getConfig() {
 	return {
 		slotsToEnable: [
 			'top_boxad',
 			'invisible_skin',
-		]
+		],
+		onInit: () => {
+			context.push('state.adStack', { id: 'invisible_skin' });
```
Comment (jacekbj) [inline extensions/wikia/AdEngine3/src/templates/roadblock-config.js:10]: It took me quite some time to understand what actually happens here :)

File: extensions/wikia/AdEngine3/src/setup.js [hunk 3]
```diff
@@ -194,7 +194,7 @@ function setupAdContext(wikiContext, isOptedIn = false, geoRequiresConsent = tru
 
 		if (slot instanceof Array) {
 			slotName = slot[0];
```
Comment (jacekbj) [inline extensions/wikia/AdEngine3/src/setup.js:196]: I just hope `slot` is not an empty array.

**Review by jacekbj — APPROVED** [review]

**Pull request merged**